### DISCUSSION TITLE
New features / message input

### DISCUSSION THREAD

**Discussion by sehaas** [AsamK/signal-cli#398] [Q&A]
I'm using signal-cli as notification service in an Home Assistant Docker container for some time now.
Since 0.7.0 it does not work any more. The HA container is based on Alpine Linux and I can't get the libzkgroup working there.
I thinks it's because of musl instead of glibc and I could not compile a static linked version (I'm not really familiar with Rust).
So I was looking for other ways to integrate signal-cli.

There are several open PRs which try to implement similar requirements (JSON from stdin or sockets, Docker support, ...).
Some of them seem to be redundant/conflicting. Are there any plans to coordinate new features like these?

For my use case I would have following requirements:
- Run in Docker container
- Accept messages via socket (RAW or even REST-API)
- Pass attachments (data and not only filenames)
(receiving messages isn't relevant for me)

I'll try do implement those features, but I don't want to push just another PR if no one needs it.
It would be great if @AsamK and others could give some feedback in which direction the development should go.

Thanks,
  sehaas

**Comment by Atomic-Bean** [comment]
Regarding attachments, singal-cli saves them into the config folder (`$HOME/.local/share/signal-cli/attachments`) so when you receive the messages they have both the original filename and the "id", they will be saved with the "id" as the filename so look for that file, copy it to the destination with the original filename.

**Reply by sehaas** [reply]
Thanks for your reply. I meant something different. The current implementation only supports Files (for sending) which are locally available. I've implemented an API which don't rely on the local filesystem, but accepts also the attachment via a Socket.

There is still some work to do on the receiving end, but I'll provide a PR in the next view days.
